### PR TITLE
generic service dashboards (emailservice, paymentservice, shippingservice)

### LINKED ISSUE #66: Create terraform for paymentservice, emailservice, and shippingservice custom dashboard
#WHAT: Create dashboard that represents four golden signals of monitoring (latency, traffic, errors, and saturation) for the three services: paymentservice, emailservice, and shippingservice

#WHY: Creates opinionated monitoring for these services

#HOW: Terraform resources added to initialization terraform scripts. Should use a terroform templatefile to provision the same dashboards for each of these services

### PR DESCRIPTION
**Pull request by nicklin37**
#WHAT: Creates dashboards for three services (emailservice, paymentservice, shippingservice) with charts with the same metrics but are filtered appropriately for each service. 

#WHY: Each service needs custom dashboard :D 

#HOW: Using Terraform template file to spin up identical dashboards for each service here

#TESTING: Added tests for dashboard creation, all tests pass. I know the file looks a little wonky, there are multiple PR's open right now that are modifying the testing script so I'll make sure it looks proper before merging this last one in.

Closes #66

### CODE REVIEW

File: terraform/monitoring/02_dashboards.tf [hunk 1]
```diff
@@ -114,4 +114,51 @@ resource "google_monitoring_dashboard" "userexp_dashboard" {
...
+      service_id = "emailservice"
+    },
+    {
+      service_name = "Shipping Service"
+      service_id = "shippingservice"
+    }
+  ]
+}
+
+# Here we iterate over the services that we defined and create a dashboard template file for each one using
```
Comment (simonz130) [inline terraform/monitoring/02_dashboards.tf:146]: nit: remove "Here we" everywhere - focus on the action: "iterate over the services..."

File: tests/monitoring_integration_test.py [hunk 2]
```diff
@@ -27,19 +27,19 @@
...
-      str -- the project name
-  """
-  project_id = os.environ['GOOGLE_CLOUD_PROJECT']
-
-  if not project_id:
-      raise MissingProjectIdError(
-          'Set the environment variable ' +
-          'GCLOUD_PROJECT to your Google Cloud Project Id.')
-  return project_id
+    """Retreieves the project id from the environment variable.
```
Comment (simonz130) [inline tests/monitoring_integration_test.py:30]: nit: Retrieves

File: tests/monitoring_integration_test.py [hunk 3]
```diff
@@ -27,19 +27,19 @@
...
+    MissingProjectIdError -- When not set.
+    Returns:
+    str -- the project name
+    """
+    project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+    if not project_id:
+        raise MissingProjectIdError(
+            'Set the environment variable ' +
+            'GCLOUD_PROJECT to your Google Cloud Project Id.')
```
Comment (simonz130) [inline tests/monitoring_integration_test.py:41]: GOOGLE_CLOUD_PROJECT

File: terraform/monitoring/02_dashboards.tf (hunk 1 above)
Comment (nicklin37) [reply terraform/monitoring/02_dashboards.tf]: Fixed :)

File: tests/monitoring_integration_test.py (hunk 2 above)
Comment (nicklin37) [reply tests/monitoring_integration_test.py]: Fixed :)

File: tests/monitoring_integration_test.py (hunk 3 above)
Comment (nicklin37) [reply tests/monitoring_integration_test.py]: Fixed! :)

**Comment by nicklin37** [comment]
Rebased and now merging

**Pull request merged**